src/wibeth/WIBEthFrameProcessor.cpp

```diff
@@ -512,11 +520,11 @@ WIBEthFrameProcessor::process_swtpg_hits(uint16_t* primfind_it, dunedaq::daqdata
 
 	  fdreadoutlibs::types::TriggerPrimitiveTypeAdapter tp;
           tp.tp.time_start = tp_t_begin;
-          tp.tp.time_peak = (tp_t_begin + tp_t_end) / 2;
-          tp.tp.time_over_threshold = int64_t(hit_tover[i]) * clocksPerTPCTick;
+          tp.tp.time_peak = tp_t_peak;
+	  tp.tp.time_over_threshold = uint64_t((hit_tover[i] - 1) * clocksPerTPCTick);
```

Comment:
I think it would be good to add a comment to explain why you have `hit_tover[i] - 1`